### ISSUE TITLE
Make it easier to add a custom TokenGranter in AuthorizationServerEndpointsConfigurer

### ISSUE THREAD

**Issue by imod** [spring-projects/spring-security-oauth#337]
This is a copy of https://jira.spring.io/browse/SEC-2707

Reference: http://stackoverflow.com/questions/25264358/spring-security-oauth2-with-custom-tokengranter-in-version-2-0

While configuring an OAuth AuthorizationServer with `AuthorizationServerEndpointsConfigurer` it is not easy to add an additional `TokenGranter`.
One is able to do something like this:

```
        List<TokenGranter> tokenGranters = getTokenGranters(); // implementation up to you
        tokenGranter = new CompositeTokenGranter(tokenGranters);
        endpoints.tokenGranter(tokenGranter);
```

but the problem is to get hold of the default `TokenGranter`s, if one gets them via `AuthorizationServerSecurityConfigurer.getTokenGranter()` then they are not initialized correctly, as e.g. `AuthorizationServerConfiguration.configure(ClientDetailsServiceConfigurer)` has not been called yet. And creating the defaults from scratch raises some other issues on how to get hold of the required references used to create these.

So this feature request asks for a proper way on how to configure/add a custom `TokenGranter` via Spring Java Config.

**Comment by bmcustodio** [comment]
@dsyer Any updates on this? Is there a workaround for the moment that doesn't involve creating and gluing the beans manually?

**Comment by dsyer** [comment]
The default token granters are not beans, but no, there has been no change since this issue was opened. It isn't really very much code to create the defaults (less then 10 lines), but I haven't tried it myself, so I don't know if there are any other obstacles. I would be surprised if there wasn't a way to defer a call to `getTokenGranter()` so that it *did* provide the defaults at the right time anyway.

**Comment by mrethers** [comment]
For now the easiest is to do as so:

```
protected static class AuthorizationServerConfiguration extends AuthorizationServerConfigurerAdapter {

...

@Override
		public void configure(AuthorizationServerEndpointsConfigurer endpoints) throws Exception {

			CompositeTokenGranter defaultTokenGranter = (CompositeTokenGranter)endpoints.getTokenGranter();
			
			List<TokenGranter> tokenGranters = new ArrayList<TokenGranter>();
			tokenGranters.add(defaultTokenGranter);
			tokenGranters.add(yourTokenGranterInstance);
			
			TokenGranter tokenGranter = new CompositeTokenGranter(tokenGranters);
			
			endpoints
					...
					.tokenGranter(tokenGranter);
		}

}

**Comment by dsyer** [comment]
Fixed in 0c0888ffb
